src/xdsl/parser.py

```diff
@@ -360,7 +360,7 @@ def parse_list(self,
         assert (len(delimiter) <= 1)
         res = list[Any]()  # Pyright do not let us use `T` here
         one = parse_optional_one()
-        if one is not None:
+        if one:
```

Comment:
We should keep the `is not None` here.
If `parse_optional_one` is parsing an integer, then it wouldn't be added to the list if it is equal to `0`.

Reply:
Makes sense.
Request: Could you help me with a minimal test for this case?
Since current codebase does not have one.

Reply:
```
parser.parse_list(parser.parse_int_literal)
```
applied to a string like `1, 0, 1` will only parse the first two values, but will return `[1]` instead of parsing everything and returning `[1, 0, 1]`

Reply:
I see. Is it easy for you to commit in this PR a test that will catch this case?

Reply:
Yes, I'll add it after my meeting!

Reply:
Sorry for the delay, it should be added now.

Reply:
Thanks!